rs/index_server/src/metrics.rs

```diff
@@ -0,0 +1,118 @@
+use std::io;
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+use futures::future::BoxFuture;
+use http_body_util::{combinators, BodyExt, Full};
+use hyper::body::{Bytes, Incoming};
+use hyper::server::conn::http1;
+use hyper::service::service_fn;
+use hyper::{Request, Response};
+use hyper_util::rt::TokioIo;
+use prometheus_client::encoding::text::encode;
+use prometheus_client::metrics::counter::Counter;
+use prometheus_client::registry::Registry;
+use tokio::net::TcpListener;
+use tokio::pin;
+use tokio::signal::unix::{signal, SignalKind};
+
+pub struct MetricsServer {
+    registry: Registry,
+    request_counter: Counter<u64>,
```

Comment:
let's make the metrics global static variable. You'd want to access it from different places. You'd also want to use `lazy_static` crate. https://www.perplexity.ai/search/how-to-create-rust-global-stat-MAaZb83BSL.HsbAwpc2siw

Try without Mutex - I think `Counter` is already `Atomic`